07-dplyr-and-databases.Rmd

```diff
@@ -0,0 +1,340 @@
+---
+title: "SQL databases and R"
+author: Data Carpentry contributors
+---
+
+```{r setup, include=FALSE}
+library(knitr)
+opts_chunk$set(
+  message = FALSE, 
+  echo = TRUE,
+  purl = FALSE
+)
+```
+
+```{r source, include=FALSE}
+source("setup.R")
+if (file.exists("portalR.sqlite")) file.remove("portalR.sqlite")
+```
+
+> ## Learning Objectives
+>
+>  By the end of this lesson the learner will:
+>
+> * Access a database from R
+> * Run a SQL query in R using dplyr.
+> * Create an SQLite database from existing .csv files.
+
+------------
+
+# Introduction
+
+So far, we have dealt with small datasets that easily fit into your computer's
+memory. But what about datasets that are too large for your computer to
+handle as a whole? In this case, storing the data outside of R and organizing it
+in a database is helpful: Connecting to the database allows you to retrieve only
+the chunks needed for the current analysis.
+
+Even better, many large datasets are already available in public or private
+databases. You can query them without having to download the data first.
+
+R can connect to almost any existing database type. For example, the
+[dplyr](https://cran.r-project.org/web/packages/dplyr/index.html) package you
+used in the previous lesson supports connecting to the popular open source
+databases [sqlite](https://sqlite.org/), [mysql](https://www.mysql.com/) and
+[postgresql](https://www.postgresql.org/), as well as google’s
+[bigquery](https://cloud.google.com/bigquery/).
+
+```{r dplyr}
+library(dplyr)
+```
+
+# Connecting to databases from R
+
+We will continue to explore the `survey` data you are
+already familiar with from previous lessons. Instead of reading the data from a 
+csv file, we will connect to an SQLite database with the same information
+(and more). The SQLite database is contained in a single file:
+`portal_mammals.sqlite` and we point R to it with dplyr's `src_sqlite()`
```

Comment:
The SQLite database is contained in a single file `portal_mammals.sqlite` that that generated during the SQL lesson. If you don't have it, you can download it from Figshare using:

```{r download-database, eval=FALSE}
## download mammals database to data folder.
download.file("https://ndownloader.figshare.com/files/2292171", "data/portal_mammals.sqlite")
```
We can point R to this database with dplyr's `src_sqlite()` command.